src/classes/action/link_action.vala

```diff
@@ -63,10 +63,17 @@ namespace pdfpc {
             return new_obj as ActionMapping;
         }
 
+        protected override void on_mouse_enter(Gtk.Widget widget, Gdk.EventMotion event) {
```

Comment:
Why do you switch visibility level for an overridden method?

Reply:
Well, I see no reason for them to be callable directly. Is it a problem?

Reply:
No, but in `ActionMapping` the virtual function is `public`.

Reply:
It's called from `View.Behaviour.PdfLink`, so it must be `public`. In general, there is quite some mess with `ActionMapping` returning copies of the class...